src/App.tsx

```diff
@@ -1,8 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import * as api from './api';
+import PostsList from './PostList';
+
+const App: React.FC = () => {
```

Comment:
Using `React.FC` is deprecated, you should either use `JSX.Element` or nothing at all (please have a look [here](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/))
```suggestion
const App: JSX.Element = () => {
```